feat(request): add request handler base

oxidia opened the pull request.

Commit:
feat(request): add request handler base

Commit:
feat(request): add default static page for testing

Commit:
feat: add parse request target as url

Commit:
fix: compile issues on mac

nerraou reviewed (commented):
Tests doesn't  work on Mac

Commit:
Fix

check for request readiness for writing first
use HTTP_MACROS
ignore body if it's GET method

nerraou reviewed (approved):
LGTM
